### DISCUSSION TITLE
DynamoDB/MosaicJSON Documentation

### DISCUSSION THREAD

**Discussion by kmarsh** [developmentseed/titiler#450] [Q&A]
👋 I'm trying to serve tiles from NAIP data in MosaicJSONs in DynamoDB (via https://github.com/kylebarron/naip-cogeo-mosaic) but can't seem to make heads or tails about how to configure TiTiler to work with this backend.

This is what I managed to cobble together, which seems to load and run OK with `uvicorn app.main:app --reload` but I have no idea what URLs, etc. to hit to make it work:

```python

from titiler.core.errors import DEFAULT_STATUS_CODES, add_exception_handlers
from titiler.mosaic.errors import MOSAIC_STATUS_CODES

from titiler.core.factory import TilerFactory
from titiler.mosaic.factory import MosaicTilerFactory

from fastapi import FastAPI
from fastapi import Query

from dataclasses import dataclass
from typing import List

from cogeo_mosaic.backends import MosaicBackend

backend = MosaicBackend("dynamodb://us-west-2/naip:2011_2013")

@dataclass
class MosaicTiler(MosaicTilerFactory):
    """Custom MosaicTilerFactory.

    Note this is a really simple MosaicTiler Factory with only few endpoints.
    """

    def register_routes(self):
        """This Method register routes to the router. """

        self.tile()
        self.tilejson()

def DatasetPathParams(url: str = Query(..., description="Dataset URL")) -> List[str]:
    """Create dataset path from args"""
    print(url)
    return url.split(",")

mosaic = MosaicTiler(reader=backend, path_dependency=DatasetPathParams)

app = FastAPI()
cog = TilerFactory()
app.include_router(mosaic.router)
add_exception_handlers(app, DEFAULT_STATUS_CODES)
add_exception_handlers(app, MOSAIC_STATUS_CODES)
```

Any help is appreciated, thanks!

**Comment by vincentsarago** [comment]
you're almost there. 

```python
def DatasetPathParams(mosaic: str = Query(..., description="mosaic name")), date: str = Query(..., description="Date")) -> str:
    """Create Mosaic URL given a name an a date."""
    return f"dynamodb://us-west-2/{mosaic}:{date}


mosaic = MosaicTiler(path_dependency=DatasetPathParams)
```

By default the `MosaicTilerFactory` will use `MosaicBackend` https://github.com/developmentseed/cogeo-mosaic/blob/master/cogeo_mosaic/backends/__init__.py#L17 to use the correct backend for a given URL. In your case the path_dependency just needs to be a string pointing to you dynamo db table. 


Note: 
- NAIP bucket is requester-pays so make sure you have those two env set: https://github.com/developmentseed/titiler-pds/blob/master/titiler_pds/routes/naip.py#L13-L15
- because of ☝️ and that you are using dynamoDB table you may want to add auth to your endpoints
